src/main/java/net/praqma/hudson/scm/ChangeLogParserImpl.java

```diff
@@ -27,10 +28,26 @@
 
 	protected static final Logger logger = Logger.getLogger( ChangeLogParserImpl.class.getName() );
 
+	private Digester createDigester(boolean secure) throws SAXException {
+		Digester digester = new Digester();
+		if (secure) {
+			digester.setXIncludeAware(false);
+			try {
+				digester.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+				digester.setFeature("http://xml.org/sax/features/external-general-entities", false);
+				digester.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+				digester.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+			} catch (ParserConfigurationException ex) {
+				throw new SAXException("Failed to securely configure xml digester parser", ex);
+			}
+		}
+		return digester;
+	}
+	
 	@Override
 	public ChangeLogSet<? extends Entry> parse( AbstractBuild build, File changelogFile ) throws IOException, SAXException {
 		List<ChangeLogEntryImpl> entries = new ArrayList<ChangeLogEntryImpl>();
-		Digester digester = new Digester2();
+		Digester digester = createDigester(true);

```

Comment:
could is by disabled with a system property?

like 

```suggestion
		Digester digester = createDigester(!Boolean.getBoolean(this.getClass().getName() + ".UNSAFE"));
```

or within the `createDigester` method?